File: FrontEnd/src/dto/Worker.ts
import type { Department } from "./Department"
import type { Experience } from "./Experience"
import type { Position } from "./Position"
import type { WorkerCompetence } from "./WorkerCompetence"
import type { WorkerTraining } from "./WorkerTraining"

export class Worker {
    id: number | null = null
    identification: string = ''
    name: string = ''
    position_id: number | null = null
    department_id: number | null = null
    recommended_id: number | null = null
    wage: number | null = null
    initial_date: Date | null = null
    type: WorkerType | null = null
    state: boolean = true
    wage_text: string = ''

    position: Position | null = null
    department: Department | null = null
    recommended: Worker | null = null

    worker_competences: WorkerCompetence[] = []
    worker_trainings: WorkerTraining[] = []
    experiences: Experience[] = []
    workers: Worker[] = []
}

File: BackEnd/db/models/worker.py
from sqlalchemy import Column, Integer, String, ForeignKey, Boolean, Date
from sqlalchemy.orm import relationship

from db.database import Base

class Worker(Base):
    __tablename__ = "Worker"

    id = Column(Integer, primary_key=True, index=True)
    identification = Column(String)
    name = Column(String)
    position_id = Column(Integer, ForeignKey("Positions.id", name='fk_worker_position_id'))
    department_id = Column(Integer, ForeignKey("Department.id", name='fk_worker_department_id'))
    recommended_id = Column(Integer, ForeignKey("Worker.id", name='fk_worker_recommended_id'))
    wage = Column(Integer)
    initial_date = Column(Date)
    type = Column(Integer)
    state = Column(Boolean)

    position = relationship("Position", back_populates="workers")
    department = relationship("Department", back_populates="workers")
    recommended = relationship("Worker", remote_side=[id], back_populates="workers")

    worker_competences = relationship("WorkerCompetence", back_populates="worker")
    worker_trainings = relationship("WorkerTraining", back_populates="worker")
    experiences = relationship("Experience", back_populates="worker")
    workers = relationship("Worker", back_populates="recommended")
    users = relationship("User", back_populates="worker")